### PR TITLE
DYN-6068 Fix packages guided tour crash.

### PR DESCRIPTION
**Pull request by reddyashish**
### Purpose

Fixes the crash in here: https://jira.autodesk.com/browse/DYN-6068

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB 

### Release Notes
Fix packages guided tour crash.

### Reviewers
@QilongTang @RobertGlobant20

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerSearchViewModel.cs [hunk 1]
```diff
@@ -1128,6 +1128,14 @@ internal IEnumerable<PackageManagerSearchElementViewModel> Search(string searchT
                 string searchTerm = searchText.Trim();
                 var packages = new List<PackageManagerSearchElementViewModel>();
 
+                //The DirectoryReader and IndexSearcher have to be assigned after commiting indexing changes and before executing the Searcher.Search() method,otherwise new indexed info won't be reflected
+                LuceneSearchUtility.dirReader = LuceneSearchUtility.writer?.GetReader(applyAllDeletes: true);
+
+                if (LuceneSearchUtility.Searcher == null && LuceneSearchUtility.dirReader != null)
+                {
+                    LuceneSearchUtility.Searcher = new IndexSearcher(LuceneSearchUtility.dirReader);
+                }
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerSearchViewModel.cs:1137]: Placing this logic inside search() function is a bit strange, is there a better place in your opinion?

**Comment by reddyashish** [comment]
The failing tests here are 52 WPFVisualizationtests and 7 NodeAutocomplete tests(which will be addressed by Roberto's PR). Are the visualization tests still failing on upstream 2.19 branch?

**Comment by QilongTang** [comment]
> The failing tests here are 52 WPFVisualizationtests and 7 NodeAutocomplete tests(which will be addressed by Roberto's PR). Are the visualization tests still failing on upstream 2.19 branch?

Yes, I think @mjkkirschner knows why they are failing on 2.19 branch but not master. Unfortunately, he is out today.

File: src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerSearchViewModel.cs (hunk 1 above)
Comment (reddyashish) [reply src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerSearchViewModel.cs:1137]: We are setting the Searcher property similarly for Node search and NodeAutocomplete search also. https://github.com/DynamoDS/Dynamo/blob/2658b00f970f487cd83926ced349e08ad5469e77/src/DynamoCore/Search/NodeSearchModel.cs#L253. We want to set this only when searcher is null. What do you think?

**Review by QilongTang — APPROVED** [review]

**Comment by QilongTang** [comment]
Pleaes cherry-pick as well

**Pull request merged**